progress/bar.go

```diff
@@ -61,59 +60,85 @@ func (b *Bar) String() string {
 		termWidth = 80
 	}
 
-	var pre, mid, suf strings.Builder
-
-	if b.message != "" {
+	var pre strings.Builder
+	if len(b.message) > 0 {
 		message := strings.TrimSpace(b.message)
 		if b.messageWidth > 0 && len(message) > b.messageWidth {
 			message = message[:b.messageWidth]
 		}
 
 		fmt.Fprintf(&pre, "%s", message)
-		if b.messageWidth-pre.Len() >= 0 {
-			pre.WriteString(strings.Repeat(" ", b.messageWidth-pre.Len()))
+		if padding := b.messageWidth - pre.Len(); padding > 0 {
+			pre.WriteString(strings.Repeat(" ", padding))
 		}
 
 		pre.WriteString(" ")
 	}
 
-	fmt.Fprintf(&pre, "%3.0f%% ", math.Floor(b.percent()))
-
-	fmt.Fprintf(&suf, "(%s/%s", format.HumanBytes(b.currentValue), format.HumanBytes(b.maxValue))
-
-	stats := b.Stats()
-	rate := stats.rate
-	if stats.value > b.initialValue && stats.value < b.maxValue {
-		fmt.Fprintf(&suf, ", %s/s", format.HumanBytes(int64(rate)))
+	fmt.Fprintf(&pre, "%3.0f%%", b.percent())
+
+	var suf strings.Builder
+	// max 13 characters: "999 MB/999 MB"
+	if b.stopped.IsZero() {
+		curValue := format.HumanBytes(b.currentValue)
+		suf.WriteString(strings.Repeat(" ", 6-len(curValue)))
+		suf.WriteString(curValue)
+		suf.WriteString("/")
+
+		maxValue := format.HumanBytes(b.maxValue)
+		suf.WriteString(strings.Repeat(" ", 6-len(maxValue)))
+		suf.WriteString(maxValue)
+	} else {
+		maxValue := format.HumanBytes(b.maxValue)
+		suf.WriteString(strings.Repeat(" ", 6-len(maxValue)))
+		suf.WriteString(maxValue)
+		suf.WriteString(strings.Repeat(" ", 7))
 	}
 
-	fmt.Fprintf(&suf, ")")
-
-	var timing string
-	if stats.value > b.initialValue && stats.value < b.maxValue {
-		timing = fmt.Sprintf("[%s:%s]", formatDuration(time.Since(b.started)), formatDuration(stats.remaining))
+	rate := b.rate()
+	// max 10 characters: ", 999 MB/s"
+	if b.stopped.IsZero() {
+		suf.WriteString(", ")
+		humanRate := format.HumanBytes(int64(rate))
+		suf.WriteString(strings.Repeat(" ", 6-len(humanRate)))
```

Comment:
I seem to be getting an error here:
```
pulling manifest 
panic: strings: negative Repeat count

goroutine 38 [running]:
strings.Repeat({0x101195a58?, 0x140004de8c0?}, 0x100d66557?)
	/opt/homebrew/Cellar/go/1.21.3/libexec/src/strings/strings.go:554 +0x2e0
github.com/jmorganca/ollama/progress.(*Bar).String(0x140000ac080)
	/Users/jmorgan/workspace/ollama/progress/bar.go:103 +0xa1c
github.com/jmorganca/ollama/progress.(*Progress).render(0x140004fe6c0)
	/Users/jmorgan/workspace/ollama/progress/progress.go:97 +0x228
github.com/jmorganca/ollama/progress.(*Progress).start(0x140004fe6c0)
	/Users/jmorgan/workspace/ollama/progress/progress.go:111 +0x74
created by github.com/jmorganca/ollama/progress.NewProgress in goroutine 1
	/Users/jmorgan/workspace/ollama/progress/progress.go:26 +0xa0
```

I'm guessing humanRate goes up to 6> characters?

Reply:
That shouldn't be possible unless rate is >999 TB/s

Reply:
Yep it happens very rarely, I re-ran `ollama pull` a few times to trigger it. It happened mid download which is strange

Reply:
I made a change fixing the divide by zero in rate() so this shouldn't be possible anymore

Reply:
Still seem to be getting it, this time on line 125

```
% go run . pull orca-mini:3b-q4_1
pulling manifest 
panic: strings: negative Repeat count

goroutine 22 [running]:
strings.Repeat({0x104c41988?, 0x140000755c0?}, 0x1048124d7?)
	/opt/homebrew/Cellar/go/1.21.3/libexec/src/strings/strings.go:554 +0x2e0
github.com/jmorganca/ollama/progress.(*Bar).String(0x14000092080)
	/Users/jmorgan/workspace/ollama/progress/bar.go:125 +0xd9c
```